### PR TITLE
Docs for StreamingProblem

### PR DESCRIPTION
**Pull request by Mobius5150**
Adds documentation for using the `StreamingProblem` class from https://github.com/microsoft/qdk-python/pull/23

### CODE REVIEW

File: articles/optimization-streaming-problem.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+The background thread will stay alive until the problem definition is finalized, and will continuously attempt to chunk terms into blocks that are individually uploaded in Azure. A chunk upload is triggered when the queued terms surpass a threshold on the number of terms, or the size of the data. This means that not every call to `add_terms()` triggers an upload.
+
+After the first call to `add_terms()` you may continue to call this method as many times as needed to add all of your terms. Note that this method is thread-safe so you may have multiple threads simultaneously adding terms to the problem definition although there is only ever one uploader thread.
+
+Once you have finished adding all of your terms, you must call the `upload()` method to finalize the problem definition (not that if you call `solver.submit(problem)`, `upload()` is called for you). The first time the `upload()` method is called the streaming problem will block on the background upload thread to complete. Once this is done, the block blob is finalized in Azure Storage and is ready to be used to solve the problem with Azure Quantum. Note that you may not add terms to a problem after it has been finalized.
```
Comment (dime10) [inline articles/optimization-streaming-problem.md:36]:
```suggestion
Once you have finished adding all of your terms, you must call the `upload()` method to finalize the problem definition (note that if you call `solver.submit(problem)`, `upload()` is called for you). The first time the `upload()` method is called the streaming problem will block on the background upload thread to complete. Once this is done, the block blob is finalized in Azure Storage and is ready to be used to solve the problem with Azure Quantum. Note that you may not add terms to a problem after it has been finalized.
```

File: articles/optimization-streaming-problem.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+Note that if `upload()` is not called on a streaming problem, the block blob in Azure Storage is never finalized. When this happens, Azure Storage will automatically delete the uploaded data within a period of time. For more information, see the [Block Blob reference](https://docs.microsoft.com/rest/api/storageservices/understanding-block-blobs--append-blobs--and-page-blobs).
+
+#### Tuning the upload
+
+Depending on the characteristics of your problem (especially density) or of your CPU/network connection you may want to tune the StreamingProblem class. There are two options for tuning the upload:
+
+- `StreamingProblem.upload_size_threshold` - the size, in bytes, of the compressed payload to upload. As terms are added they are compressed one the fly. When the size of the staged compressed payload surpasses this threshold the chunk is uploaded.
```
Comment (dime10) [inline articles/optimization-streaming-problem.md:44]:
```suggestion
- `StreamingProblem.upload_size_threshold` - the size, in bytes, of the compressed payload to upload. As terms are added they are compressed on the fly. When the size of the staged compressed payload surpasses this threshold the chunk is uploaded.
```

File: articles/optimization-streaming-problem.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+- `StreamingProblem.upload_terms_threshold` - the threshold for the number of terms that trigger an uploaded. When the number of queued terms exceed this threshold an upload is triggered. If your problem has high connectivity (terms have many variables) you may choose to lower this threshold.
```
Comment (dime10) [inline articles/optimization-streaming-problem.md:45]:
```suggestion
- `StreamingProblem.upload_terms_threshold` - the threshold for the number of terms that trigger an upload. When the number of queued terms exceed this threshold an upload is triggered. If your problem has high connectivity (terms have many variables) you may choose to lower this threshold.
```

**Review by dime10 — COMMENTED** [review]
Thank you @Mobius5150 for contributing this.

File: articles/optimization-streaming-problem.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+In general:
+
+- If you have plenty of free memory, consider _increasing_ the thresholds
+- If you have a slow network connection, consider _increasing_ the thresholds
+- If you are highly memory constrained, consider _decreasing_ the size threshold
+
+#### Statistics of the uploaded problem
+
+The `StreamingProblem` class calculates some statistics about the generated problem as terms are added. To see information about the terms already added to the problem, access the `stats` member of the class which is a dictionary of calculated statistics and contains the following information:
```
[the remark this answers is not in the document]
Comment (Mobius5150) [reply articles/optimization-streaming-problem.md:63]: Added

**Review by frtibble — APPROVED** [review]

**Review by dime10 — APPROVED** [review]

**Pull request merged**